src/Uno.UI.Runtime.Skia.Wpf/Input/WpfKeyboardInputSource.cs

```diff
@@ -116,10 +116,26 @@ private static uint GetScanCode(KeyEventArgs args)
 		}
 	}
 
-	private static char? KeyCodeToUnicode(uint keyCode)
+	private static char? KeyCodeToUnicode(uint keyCode, VirtualKey virtualKey)
 	{
 		var result = InputHelper.WindowsScancodeToUnicode(keyCode);
-		return result.Length > 0 ? result[0] : null; // TODO: supplementary code points
+
+		// WindowsScancodeToUnicode doesn't pick up Numpad keys
+		return virtualKey switch
+		{
+			VirtualKey.NumberPad0 => '0',
+			VirtualKey.NumberPad1 => '1',
+			VirtualKey.NumberPad2 => '2',
+			VirtualKey.NumberPad3 => '3',
+			VirtualKey.NumberPad4 => '4',
+			VirtualKey.NumberPad5 => '5',
+			VirtualKey.NumberPad6 => '6',
+			VirtualKey.NumberPad7 => '7',
+			VirtualKey.NumberPad8 => '8',
+			VirtualKey.NumberPad9 => '9',
+			VirtualKey.Decimal => '.',
```

Comment:
Should this then be part of `WindowsScancodeToUnicode` itself? And why WindowsScancodeToUnicode doesn't work correctly?

Reply:
It could, but I would prefer that `WindowsScancodeToUnicode` be kept to the bare minimum of interacting with the system kernel.